Cocina/Cocina/Settings/EditProfileViewController.swift

```diff
@@ -90,10 +127,98 @@ extension EditProfileViewController: UITableViewDataSource, UITableViewDelegate
+        cell.textField.tag = indexPath.row
+        if cell.textField.tag == 3 {
+            cell.textField.isSecureTextEntry = true
+        }
         return cell
     }
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
     }
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        // update the user object
+        textField.resignFirstResponder()
+        print(textField.tag)
+        return true
+    }
+}
+
+// MARK: EditProfileHeaderDealagte & PickerVC 
+extension EditProfileViewController: EditProfileHeaderDelegate, PHPickerViewControllerDelegate {
+    func didChangeProfileHeaderImage(to: UIImage) {
+        
+    }
+    
+    func didTapEditImageButton() {
+        print("Show image picker")
+        showImagePickerOptions()
+    }
+    
+    // returns an image picker with source type
+    func imagePicker(sourceType: UIImagePickerController.SourceType) -> UIImagePickerController {
+        let imagePicker = UIImagePickerController()
+        imagePicker.sourceType = .camera
+        return imagePicker
+    }
+    
+    func showImagePickerOptions() {
+        // alert view controller
+        let alertVC = UIAlertController(title: "Chose your profile picture", message: "Choose a photo or customize with icon", preferredStyle: .actionSheet)
+        
+        // image picker for camera
+        let cameraAction = UIAlertAction(title: "Camera", style: .default) { [weak self] (action) in
+            guard let self = self else { return }
+            let cameraImagePicker = self.imagePicker(sourceType: .camera)
+            self.present(cameraImagePicker, animated: true)
+            
+        }
+        
+        //image picker for library using PHPicker instead of uiimage library
+        let libraryAction = UIAlertAction(title: "Photo Library", style: .default) { [weak self] (action) in
+            guard let self = self else { return }
+            
+            // configurations for picker
+            var config = PHPickerConfiguration()
+            config.filter = .images
+            config.selectionLimit = 1
+            config.selection = .default
+            let picker = PHPickerViewController(configuration: config)
+            picker.delegate = self
+            self.present(picker, animated: true)
+        }
+        
+        // cancel
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertVC.addAction(cameraAction)
+        alertVC.addAction(libraryAction)
+        alertVC.addAction(cancelAction)
+        self.present(alertVC, animated: true)
+    }
+    
+    // The picker does not dismiss when this method is called so you need to dismiss the picker yourself.
+    func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
+        
+        if !results.isEmpty {
+            let results = results.first!
```

Comment:
Let's try to use if let or other forms of unwrapping optional values instead of force unwrapping. I see you're checking that the `results` are not empty, so technically, force unwrapping is fine here, but it is better for production code to use optional unwrapping because crazy things happen in the wild. In the event that this did not work for some crazy reason, it would crash the app. I doubt that this would happen since this is Apple's delegate but just to be safe.